src/subscription.ts

```diff
@@ -216,14 +257,37 @@ export default class extends Deployed {
 
   /**
    * @remarks
-   * Get given Allowance amount.
-   *
-   * @param a - address
-   *
+   * Get given User balance.
+   * @param u - user address
+   * @param t - token address
    */
-  async getUserBalance(a: string): Promise<any> {
-    const wei = await this.erc20Contract?.functions.balanceOf(a)
-    return this.vendor.convertWeiToEth(wei, this.tokenPrecision || 18)
+  async getUserTokenBalance(u: string, t: string): Promise<any> {
```

Comment:
don't remove the `getUserBalance` method but do add the `getUserTokenBalance()`
Also, we don't need the token address here since we are already taking it via subscriptionAt function, so derive it from this.erc20Contract